### PR TITLE
Refactor event command

### PR DESCRIPTION
**Pull request by TChan92**

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -13,11 +13,11 @@ arrow = "==0.10.0"
 alembic = "==1.0.0"
 raven = "*"
 psutil = "*"
-pyyaml = "*"
+PyYAML = "*"
 dblpy = "==0.1.6"
 
 [dev-packages]
-pylint = "*"
+black = "==19.3b0"
```
Comment (jgayfer) [inline Pipfile]: I'd prefer if we make this change in a separate PR as it's not related to the refactor.

File: apollo/input/event_time_zone_input.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+            resp = (await self.bot.get_next_pm(ctx.author)).content
+            if self._valid_time_zone_input(resp):
+                time_zone_index = int(resp) - 1
+                if time_zone_index in range(len(ISO_TIME_ZONES)):
+                    return ISO_TIME_ZONES[time_zone_index]
+            else:
+                await ctx.author.send(t("event.invalid_time_zone"))
+
+    def _valid_time_zone_input(self, value):
+        return value.isdigit() and int(value) in range(1, len(ISO_TIME_ZONES) + 1)
```
Comment (jgayfer) [inline apollo/input/event_time_zone_input.py]: We already have a time zone input service. What's the thinking behind adding another one?

File: apollo/input/start_time_input.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+from apollo.translate import t
+
+
+class StartTimeInput:
+    def __init__(self, bot):
+        self.bot = bot
+
+    async def call(self, ctx, iso_time_zone, update=False):
+        """Retrieve a datetime UTC object from the user"""
+        await ctx.author.send(t("event.start_time_prompt"))
```
Comment (jgayfer) [inline apollo/input/start_time_input.py]: I'd prefer if we kept the message prompt separate from these input services. By including a message, it's make these services hard to reuse. 

For example, what if I wanted to get a start time outside of a direct message? I wouldn't be able to reuse this service without first modifying it.

Our existing time zone input is nice as it takes a channel as a parameter, allowing it to be used in more than one context.

File: apollo/permissions.py [hunk 4]
```diff
@@ -3,6 +3,7 @@
 
 class HavePermission:
     """Checks whether a user has the necessary permission to perform an action"""
+
```
Comment (jgayfer) [inline apollo/permissions.py]: Fairly minor, would would be nice to have unrelated linting changes in a separate PR/commit.

File: Pipfile (hunk 1 above)
Comment (TChan92) [reply Pipfile]: Good point, I'll change that.

File: apollo/permissions.py (hunk 4 above)
Comment (TChan92) [reply apollo/permissions.py]: Will do

File: apollo/input/event_time_zone_input.py (hunk 2 above)
Comment (TChan92) [reply apollo/input/event_time_zone_input.py]: I added this file because this one reads pms instead of messages.
I was also going to make it accept None when the user is editing an event.
I also had the prompt inside, but will change that.

File: apollo/input/start_time_input.py (hunk 3 above)
Comment (TChan92) [reply apollo/input/start_time_input.py]: Would it make more sense to have the prompt outside the class entirely, like back in EventCommand or to have the input class have a prompt function?

Comment (jgayfer) [reply apollo/input/start_time_input.py]: Yes, I think so. It would be more in line with the SRP (single responsibility principle).

Error messages aside, we mostly want to create a service that reads input from the user. I would argue that prompting the user for said message would be an additional responsibility that we could provide elsewhere.
